### PR TITLE
feat: Approval list is sorted in reverse chronological order

### PR DESCRIPTION
**Pull request by shuofan**
#### What this PR does / why we need it:
Approval list is sorted in reverse chronological order

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](paste your link here)

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |   Approval list is sorted in reverse chronological order           |
| 🇨🇳 中文    |   审批列表按时间倒序排列           |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: modules/admin/component-protocol/components/personal-workbench/messageList/approval.go [hunk 1]
```diff
@@ -39,6 +41,17 @@ func (l *MessageList) doFilterApproval() (data *list.Data) {
 		UserIDs: append(blockUsers, deployUsers...),
 	}
 
+	sort.Slice(data.List, func(i, j int) bool {
```
Comment (sfwn) [inline modules/admin/component-protocol/components/personal-workbench/messageList/approval.go:44]: it's wrong if you have more than 1000 db records.
you should implement in db side.

**Review by sfwn — DISMISSED** [review]

File: modules/admin/component-protocol/components/personal-workbench/messageList/approval.go (hunk 1 above)
Comment (shuofan) [reply modules/admin/component-protocol/components/personal-workbench/messageList/approval.go:44]: Yes, we may union tables or refactor approval function. 
it works as there are less than 300 records in db.

Comment (Effet) [reply modules/admin/component-protocol/components/personal-workbench/messageList/approval.go:44]: then, at least leave some TODO

File: modules/admin/component-protocol/components/personal-workbench/messageList/approval.go [hunk 2]
```diff
@@ -39,6 +41,17 @@ func (l *MessageList) doFilterApproval() (data *list.Data) {
...
+		t1, ok := data.List[i].Extra.Extra["updatedAt"].(time.Time)
+		if !ok {
+			return false
+		}
+		t2, ok := data.List[j].Extra.Extra["updatedAt"].(time.Time)
```
Comment (Effet) [inline modules/admin/component-protocol/components/personal-workbench/messageList/approval.go:49]: encapsulate a func like `getUpdateAt`, and check nil if needed

**Review by Effet — CHANGES_REQUESTED** [review]

**Review by Effet — APPROVED** [review]

**Pull request merged**